tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3767 @@
+                        </blockquote>
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 2</strong> to denote a <i>Metadata Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data Length</td>
+                                <td>uint32</td>
+                                <td>
+                                    Length of the following data field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data</td>
+                                <td>ByteArray</td>
+                                <td>Binary data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                </td>
+            </tr>
+        </table>
+    </details>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+#### Output Syntactic Validation
+
+- `Amount` field must fulfill the dust protection requirements and must not be `0`.
+- `Native Tokens Count` must not be greater than `Max Native Tokens Count`.
+- `Native Tokens` must be lexicographically sorted based on `Token ID`.
+- Each <i>Native Token</i> must be unique in the set of `Native Tokens` based on its `Token ID`. No duplicates are
+  allowed.
+- `Amount` of any <i>Native Token</i> must not be `0`.
+- It must hold true that `Unlock Conditions Count` = `2`.
+- `Unlock Condition Type` of an <i>Unlock Condition</i> must define one of the following types:
+  - <i>State Controller Address Unlock Condition</i>
+  - <i>Governor Address Unlock Condition</i>
+- <i>Unlock Conditions</i> must be sorted in ascending order based on their `Unlock Condition Type`.
+- Syntactic validation of all present unlock conditions must pass.
+- It must hold true that `0` ≤ `Blocks Count` ≤ `3`.
+- `Block Type` of a <i>Block</i> must define on of the following types:
+  - <i>Sender Block</i>
+  - <i>Issuer Block</i>
+  - <i>Metadata Block</i>
+- Syntactic validation of all present feature blocks must pass.
+- When `Alias ID` is zeroed out, `State Index` and `Foundry Counter` must be `0`.
+- `State Metadata Length` must not be greater than `Max Metadata Length`.
+- `Address` of <i>State Controller Address Unlock Condition</i> and `Address` of
+  <i>Governor Address Unlock Condition</i> must be different from the alias address derived from `Alias ID`.
```

Comment:
Maybe my understanding but it's not clearly stated that they are both mandatory

Reply:
Actually, except for Extended, it's not clear there must be an address/state/governor

Reply:
Just saw it's there for NFT sorry 😬

Reply:
> Actually, except for Extended, it's not clear there must be an address/state/governor

Implicitly it is, as it is defined for the alias output that
 -  supported unlock conditions are the state/governor,
 - `Unlock Conditions Count` must be `2`,
 - and there can be at most one of each type of unlock conditions in the output.
 
 This means ther ehas to be one state controller unlock condition and one governor in each alias otuput, otherwise they are syntactically invalid.

Reply:
Yeah right, a bit of mental overhead but that's correct, thanks 👍